include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -313,6 +324,90 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
   return overallObjective;
 }
 
+// Transforms the candidate into the given bounds.
+template<typename SelectionPolicyType>
+template<typename BaseMatType>
+void CMAES<SelectionPolicyType>::BoundaryTransform(BaseMatType& matrix)
+{
+  typedef typename BaseMatType::elem_type ElemType;
+  const double diff = (upperBound - lowerBound) / 2.0;
+  const double al = std::min(diff, (1 + std::abs(lowerBound)) / 20.0);
+  const double au = std::min(diff, (1 + std::abs(upperBound)) / 20.0);
+  const double xlow = lowerBound - 2 * al - diff;
```

Comment:
Do you think we should make t possible to define bounds for all coordinates similar to what has been done for the PSO method: 

https://github.com/mlpack/ensmallen/blob/70a2be8646034300cf884649f7de8f5cc9eff75d/include/ensmallen_bits/pso/pso.hpp#L91-L92

It uses a 1x1 matrix as a default, so the current implementation but you use a matrix of the same size as the coordinates as well. Let me know what you think. We can also implement the change another time and merge this as is.

Reply:
Hi @zoq, I implemented this boundary handing function after referring the N Hansen's implementation and in my opinion, I think we should merge this as it is and if there are any issues in the future I will try to change this accordingly. Thanks.

Reply:
Okay, fine for me.